packages/shared/utils/types.ts

```diff
@@ -1,4 +1,6 @@
-import type { ComputedRef, Ref, WatchOptions, WatchSource } from 'vue-demi'
+import type { ComputedRef, MaybeRef, Ref, WatchOptions, WatchSource } from 'vue-demi'
+
+export type { MaybeRef, MaybeRefOrGetter } from 'vue-demi'
```

Comment:
I don't think this is gonna work for Vue 2 since Vue 2 doesn't expose `MaybeRef` and `MaybeRefOrGetter` types. Which is why `vueuse` is maintaining its own version of these types.

We should probably
1. Update the implementations here to match the type in 3.5
2. Change `useEventSource/index.ts` to import from `@vueuse/shared` instead of `vue-demi`